### ISSUE TITLE
Dose not work inputting the numeric keypad

### ISSUE THREAD

**Issue by akiyosi** [akiyosi/goneovim#120]
Input using the numeric keypad does not work.

**Comment by trullooo** [comment]
Hi akiyosi, I have the same problem... Is there a fix for this? Am I missing some settings?
Thx

**Comment by akiyosi** [comment]
@trullooo 
hi
I don't have a numeric keyboard so it's hard to investigate this.
I have created a branch to investigate this. 
Could you please get the investigation binary below and see what you get when you type in the numeric keyboard?

https://github.com/akiyosi/goneovim/actions/runs/342857468

**Comment by trullooo** [comment]
Hi akiosi, thanks for your replay.
I just tried version Goneovim-latest-linux-5.15.1.
I open it, I type A just to shift to INSERT mode; letters typing works, but
still nothing for the numeric keypad.
This is what i see on the terminal while typing:
./goneovim
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
QBasicTimer::start: QBasicTimer can only be used with threads started with
QThread
DEBUG:: a 65 0 a
DEBUG:: 7 55 536870912 <7>
DEBUG:: 8 56 536870912 <8>
DEBUG:: 9 57 536870912 <9>

Thanks a lot
trullooo

**Comment by akiyosi** [comment]
Thanks for confirming :D 
I have fixed the code that I believe is relevant to this issue. Could you please give it a try?
https://github.com/akiyosi/goneovim/actions/runs/343947162

**Comment by trullooo** [comment]
Hi,
with this version numeric keypad works!! Good job.

In case you want some more debugging,
1) I noticed that the autosave function, which I set in
.config/nvim/init.vim with
au FocusLost * silent! wa
does not work inside goneovim.
2) I can change only the Font size, but not the font itself which is not
the same I use in nvim (Ubuntu Mono Regular) and in terminal.
3) The shortcut Ctrl+Shift+v for Paste command does not work.
I hope to be helpful with my comments,
Thx for your GUI, really good
trullooo

**Comment by akiyosi** [comment]
@trullooo 
Thanks for providing the information that could fix this problem. :)

> 1) I noticed that the autosave function, which I set in
> .config/nvim/init.vim with
> au FocusLost * silent! wa
> does not work inside goneovim.

Yes, this is currently not working.
I now make an issue about this. I will fix soon.
https://github.com/akiyosi/goneovim/issues/178

> 2) I can change only the Font size, but not the font itself which is not
> the same I use in nvim (Ubuntu Mono Regular) and in terminal.

Sorry, I don't quite understand the problem. Please give us some more details.

> 3) The shortcut Ctrl+Shift+v for Paste command does not work.

See the following.
https://github.com/akiyosi/goneovim/issues/163#issuecomment-695770718

**Comment by trullooo** [comment]
Hi all, sorry, I guess I have been not clear.
In neovim (and in terminal too) I have the Ubuntu Mono Regular font. When I
open Goneovim the font is not the same. If I modify the setting.toml file
to change the font, nothing happens.
I can modify just the font size, but not the font itself.
Hope I have been clear.
Thx
trullooo

**Comment by akiyosi** [comment]
> In neovim (and in terminal too) I have the Ubuntu Mono Regular font. When I
> open Goneovim the font is not the same. If I modify the setting.toml file
> to change the font, nothing happens.
> I can modify just the font size, but not the font itself.
> Hope I have been clear.

Ah, sorry for the confusing documentation.
The fonts in `setting.toml` basically specify the fonts used in the external UI of the application and the default fonts.
How about setting the following?

```
set guifont=Ubuntu\ Mono\ Regular:h12
```

**Comment by trullooo** [comment]
Hi all,
I tried but it seems I cannot change the font in the editor, only the
external UI font (for me is not a big problem).
Other things I noticed: the numeric keypad works now but some keys don't:
the "-", the "+" and the "." are not working. All the other keys work.
If I double click on a file .py in the SideBar the file gets opened in
another text editor (kate in my case).Are there shortcuts to move the focus
on the workspace and back on the editor?
Thx a lot
trullooo

**Comment by akiyosi** [comment]
> I tried but it seems I cannot change the font in the editor, only the
> external UI font (for me is not a big problem).

Hmmm, how and when exactly did you set it up? 

> Other things I noticed: the numeric keypad works now but some keys don't:
> the "-", the "+" and the "." are not working. All the other keys work.

Could you show me the standard output of the previous key input debugging app when you input the key that causes the problem?

https://github.com/akiyosi/goneovim/actions/runs/342857468

> If I double click on a file .py in the SideBar the file gets opened in another text editor (kate in my case).Are there shortcuts to move the focus on the workspace and back on the editor?

There are two key things that make this possible.

The first point: the application that opens when you click on an item in the sidebar is the one associated with your system. I believe your system is configured to launch Kate when you click on a python file.

Second point: on your system, even if you set the standard editor for python files to goneovim, a different instance of goneovim will be launched when you actually click on it. If you want to open this in an existing instance of goneovim, you need to use [nvr](https://github.com/mhinz/neovim-remote). The following article describes the relevant techniques. For the configuration of the Python file association to the system, you need to use `nvr` to set up an existing instance of goneovim with the path information of the target python file and an open command to send it to the existing instance of goneovim.

https://github.com/akiyosi/goneovim/wiki/Technique#integration-with-nvr

**Comment by trullooo** [comment]
Hi all,
this is what I set in the setting.toml file:
FontFamily = "Monospace"
# Font = "Ubuntu Mono Regular"
FontSize = 11
In the init.vim file I don't put anything and the terminal font is set to
UbuntuMonoREgular. The font I see in neovim is UbuntuMono, the one I see in
goneovim is different

This is what I get on the terminal when I type "-", "+" and "." in the
numeric keypad
DEBUG:: - 45 536870912 <Minus>
DEBUG:: + 43 536870912 <Plus>
DEBUG:: . 46 536870912 <Point>
Nothing appears in the editor window.
I will dive into the nvr to understand and fix the last issue.
Thx a lot
trullooo

**Comment by akiyosi** [comment]
> this is what I set in the setting.toml file:
> FontFamily = "Monospace"
> \# Font = "Ubuntu Mono Regular"
> FontSize = 11
> In the init.vim file I don't put anything and the terminal font is set to
> UbuntuMonoREgular. The font I see in neovim is UbuntuMono, the one I see in
> goneovim is different

Please put the following into init.vim 

```
set guifont=fontname:h12
```

> This is what I get on the terminal when I type "-", "+" and "." in the
> numeric keypad
> DEBUG:: - 45 536870912 <Minus>
> DEBUG:: + 43 536870912 <Plus>
> DEBUG:: . 46 536870912 <Point>
> Nothing appears in the editor window.
> I will dive into the nvr to understand and fix the last issue.

Thanks :)
The issue will be fixed in fb84883
